src/visualization/plot_recipes.jl

```diff
@@ -569,13 +586,6 @@ function Base.getindex(pd::PlotData1D, variable_name)
 end
 
 Base.eltype(pd::PlotData1D) = Pair{String, PlotDataSeries1D}
-function Base.iterate(pd::PlotData1D, state=1)
-  if state > length(pd)
-    return nothing
-  else
-    return (pd.variable_names[state] => pd[pd.variable_names[state]], state + 1)
-  end
-end
```

Comment:
Why did you remove this?

Reply:
It was redundant - the `Base.iterate(pd::PlotData2D, state=1)` iterator does the same thing, so I just redefined the signature `Base.iterate(pd::AbstractPlotData, state = 1)`. 

I can revert the changes if we feel like we should keep it.

Reply:
I see - that's good and we can keep your version :+1: